packages/backend/src/open_payments/invoice/routes.ts

```diff
@@ -50,12 +50,15 @@ async function getInvoice(
   const invoice = await deps.invoiceService.get(invoiceId)
   if (!invoice) return ctx.throw(404)
 
-  const amountReceived = await deps.accountingService.getBalance(invoice.id)
-  if (amountReceived === undefined) {
+  const balance = await deps.accountingService.getBalance(invoice.id)
+  if (balance === undefined) {
     deps.logger.error({ invoice: invoice.id }, 'balance not found')
     return ctx.throw(500)
   }
 
+  const amountReceived = invoice.amountToReceive
+    ? invoice.amountToReceive + POSITIVE_SLIPPAGE - balance
+    : balance
```

Comment:
I'm wondering if we should have a function in the invoice service rather to account for this, if somewhere else wants to quote the invoice balance, they will need to know this has to happen

Reply:
I started to add a function yesterday and then I added `getTotalReceived` to the `accountingService`. https://github.com/interledger/rafiki/pull/199/commits/2ce777887690604405c2dea93b5f5354b6aa6708
I'll put that in this pr.